src/diagnostics/core_diagnostics.jl

```diff
@@ -733,3 +733,372 @@ add_diagnostic_variable!(
+        cloud_cover .=
+            ifelse.(
+                cache.precomputed.cloud_diagnostics_tuple.cf .> 0.0f0,
+                1.0f0,
+                0.0f0,
+            )
+        Operators.column_integral_definite!(out, cloud_cover)
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "clvi",
+    long_name = "Vertical Cloud Fraction Integral",
+    standard_name = "vertical_cloud_fraction_integral",
+    units = "m",
+    comments = """
+    The total height of the column occupied at least partially by cloud.
+    """,
+    compute! = compute_clvi!,
+)
+
+
+###
+# Total Column Integrated Specific Humidity (2d)
+###
+compute_prw!(out, state, cache, time) =
+    compute_prw!(out, state, cache, time, cache.atmos.moisture_model)
+compute_prw!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("prw", model)
+
+function compute_prw!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        Operators.column_integral_definite!(out, state.c.ρq_tot)
+        return out
+    else
+        Operators.column_integral_definite!(out, state.c.ρq_tot)
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "prw",
+    long_name = "Water Vapor Path",
+    standard_name = "atmospheric_mass_content_of_water_vapor",
+    units = "kg m^-2",
+    comments = "Vertically integrated specific humidity",
+    compute! = compute_prw!,
+)
+
+###
+# Column integrated relative humidity (2d)
+###
+compute_hurvi!(out, state, cache, time) =
+    compute_hurvi!(out, state, cache, time, cache.atmos.moisture_model)
+compute_hurvi!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("hurvi", model)
+
+function compute_hurvi!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        saturation_vi = zeros(axes(Fields.level(state.f, half)))
+        water_vapor_saturation = cache.scratch.ᶜtemp_scalar
+        @. water_vapor_saturation =
+            state.c.ρ *
+            TD.q_vap_saturation.(thermo_params, cache.precomputed.ᶜts)
+        Operators.column_integral_definite!(
+            saturation_vi,
+            water_vapor_saturation,
+        )
+        q_vap = cache.scratch.ᶜtemp_scalar
+        q_vap .=
+            TD.vapor_specific_humidity.(
+                CAP.thermodynamics_params(cache.params),
+                cache.precomputed.ᶜts,
+            )
+        saturation = cache.scratch.ᶜtemp_scalar
+        @. saturation = state.c.ρ * q_vap / saturation_vi
+        Operators.column_integral_definite!(out, saturation)
+        return out
+    else
+        saturation_vi = zeros(axes(Fields.level(state.f, half)))
```

Comment:
You are allocating a new field here. You can use scratch space to remove this allocation.

Reply:
Noting that we are keeping the allocation here after offline discussion where we discovered that scratch does not have a correctly sized field for vertical integrals